### ISSUE TITLE
Add support for VxWorks OS

### ISSUE THREAD

**Issue by wenbin0301** [openssl/openssl#11083]

**Comment by levitte** [comment]
It's already there, isn't it?  #11084 (now closed) *removed* the existing support, which doesn't seem like a good way to add support.  Perhaps modifying the existing support with what you think it should be today?

**Comment by wenbin0301** [comment]
Actually the VxWorks relevant code and configurations existing in current OpenSSL project has been not maintained for a long time. They are outdated and useless. We are planning to upstream our latest change and that is why I opened a new feature request #11083. The whole changes will not be trivial and we will sign a CLA on behalf of WindRiver before we commit them.

One question, is there any CI(such as jekins, buildbot) to run before committing a code change?

Thanks,
Wenbin

**Comment by levitte** [comment]
> One question, is there any CI(such as jekins, buildbot) to run before committing a code change?

I'm slowly working on a buildbot setup using buildbot_travis.  There are some quirks I'm working through.  So for the moment, no, but coming up

**Comment by wenbin0301** [comment]
Thank you for your reply, Richard.

If there was no CI, I’d like to know how to ensure my commit doesn’t break existing test? Furthermore, is there any guideline about how to add/integrate test support for a new OS that is to ensure other people’s commit doesn’t break the function on VxWorks?

Thanks,
Wenbin

**Comment by levitte** [comment]
> If there was no CI, I’d like to know how to ensure my commit doesn’t break existing test?

VxWorks is one of those platforms that we classify as community provided, i.e. someone out there takes the responsibility to ensure this.  That being said, `make test` runs a whole package of diverse tests using the perl testing framework (Test::More) and diverse test programs (written in C), so if you can run those with no error, you have done your part, and until there are buildbot workers demonstrating this, we will simply have to trust you.

> Furthermore, is there any guideline about how to add/integrate test support for a new OS that is to ensure other people’s commit doesn’t break the function on VxWorks?

Try to write the tests in a platform agnostic way.  We do rely a lot on perl for that, as it is, as far as I know, *the* script language that covers the most of what's out there, and we rely on File::Spec to produce platform specific paths (we have a lot of functionality in our module OpenSSL::Test, which wraps around Test::More)

Does that help?

**Comment by mattcaswell** [comment]
> If there was no CI,

Just to clarify, there *is* CI. Its just not something you can hook into prior to submitting a PR. The CIs are run on all submitted PRs automatically. Prior to submitting a PR most people just run `make test`.

**Comment by levitte** [comment]
@mattcaswell, which one of our current CIs test specifically for VxWorks?  I do believe that's part of the question asked.

**Comment by wenbin0301** [comment]
Please see my comments inline.

Thanks,
Wenbin

**Comment by levitte** [comment]
> There is one general question. For which platforms do contributors have to run the test(by make test) before their commit?  If not for all platforms, seems it is still very possible one commit breaks the test for other platforms. The key issue is it is impossible for contributors to have all platforms available. So I thought if there was CI, we could provide a Jenkins link dedicated for VxWorks platform to be integrated into it.
> 
> Does that help?

Yes, and if that can be a buildbot worker just as well, it should work with the buildbot I'm working on.  I really need to press on with that...

**Comment by wenbin0301** [comment]
So what platforms does current CI run for? Is it possible to add CI test for VxWorks? What should I do to support that?

**Comment by mattcaswell** [comment]
> @mattcaswell, which one of our current CIs test specifically for VxWorks? I do believe that's part of the question asked.

Right - there is no CI for VxWorks...but there is CI. It wasn't entirely clear to me from the question whether this part was specifically VxWorks.

**Comment by wenbin0301** [comment]
Thank you @mattcaswell and sorry for my unclear question.

Basically my question left now is what should I do to add CI specifically for VxWorks. Richard said he is working on buildbot. Hope there will be detailed step description about how to add the CI for a new OS platform then.

Thanks,
Wenbin

**Comment by mattcaswell** [comment]
> Basically my question left now is what should I do to add CI specifically for VxWorks. Richard said he is working on buildbot. Hope there will be detailed step description about how to add the CI for a new OS platform then.

There is nothing to be done right now. Our CIs are currently based on Travis and AppVeyor which (AFAIK) don't have VxWorks support. Once buildbot is set up it will enable people to set up platform specific workers. But until then we can't do VxWorks as part of our existing CIs.

**Comment by nhorman** [comment]
VxWorks, being on our list of unadopted platforms is not something that will get planned.  However @wenbin0301 if you are still interested in becoming a community maintainer for this issue, please open a PR against https://github.com/openssl/general-policies adding yourself as the VxWorks maintainer for review.

For now though
Marking as inactive, to be closed at the end of 3.4 dev barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.
